widgetry/src/event.rs

```diff
@@ -51,24 +51,45 @@ impl Event {
                 position.to_logical(scale_factor).into(),
             )),
             WindowEvent::MouseWheel { delta, .. } => match delta {
+                // "In the beginning" a spinnable mouse wheel was the only input hardware for
+                // scrolling. Each "increment" of the mouse wheel indicated that the application
+                // should scroll one line of text.
+                //
+                // Since the advent of touchpads and tablets, much finer grained scrolling is used,
+                // and consequently some input systems started expressing scroll distances from such
+                // input devices in "pixels" rather than "lines". 
+                //
+                // However some backends (e.g. x11) will express all scrolling as `LineDelta` — on
+                // those systems, touchpad drags will simply be scaled to some presumed equivalent
+                // number of lines.
+                //
+                // Widgetry expresses all scrolling in terms of MouseWheelScroll, which uses "Lines".
+                //
+                // Anymore "a line" usually doesn't correspond to "a literal line of text" in the
+                // application, and must usually just be considered an abstract unit of
+                // measurement, because of things like configurable scroll sensitivity, variable
+                // text size, and the afforementioned advent of touchpads.
+                //
+                // With "Reverse" scrolling, positive values indicate upward or rightward scrolling.
+                // With "Natural" scrolling, it's the opposite.
                 MouseScrollDelta::LineDelta(dx, dy) => {
                     if dx == 0.0 && dy == 0.0 {
                         None
                     } else {
-                        Some(Event::MouseWheelScroll(
-                            scroll_wheel_multiplier() * f64::from(dx),
```

Comment:
Looks like the mouse_wheel_multiplier stuff is no longer necessary. A similar scroll inversion bug for web was recently fixed here (since 0.22.2): https://github.com/rust-windowing/winit/pull/1665

Big take away for me is that input muxing libraries must be a PITA to write/maintain - essentially reverse-engineering 1,000 arbitrary decisions for a dozen platforms.

Reply:
@dabreegster - would you mind testing that hscroll and vscroll work in the proper direction for you on web?

Reply:
Both work on web and native!